### PR TITLE
Accept hostname in `ASM::Util.get_preferred_ip`

### PR DESCRIPTION
**Pull request by gavin-scott**
The vmware module uses this method to determine the local IP to use as
an NFS mount. In cases where the ESXi hosts are added to the vcenter
via their hostname rather than IP this was failing.

### CODE REVIEW

**Comment by gavin-scott** [comment]
@sushilrai this is a miscellaneous issue I ran into yesterday.

**Comment by sushilrai** [comment]
There is a styling error.

File: lib/asm/util.rb [hunk 1]
```diff
@@ -83,30 +83,44 @@ def self.first_host_ip
...
+    #
+    # @return [String] the local host IP address
+    def self.get_preferred_ip(remote_host, logger=nil)
       tries = 0
       max_tries = 10
       preferred = nil
 
+      ip = Resolv.getaddress(remote_host)
+
       while tries < max_tries
```
Comment (gavin-scott) [inline lib/asm/util.rb]: @sushilrai any idea why we're doing retry here? I would think that short of adding or removing a network interface that the return value of `ip route get` wouldn't change.

Comment (sushilrai) [reply lib/asm/util.rb]: I dont recall this, but possible scenario could be network interface not accessible while resolving the IP address

**Pull request merged**